src/shm/mod.rs

```diff
@@ -49,6 +50,12 @@ impl ShmState {
     }
 }
 
+impl ProvidesBoundGlobal<wl_shm::WlShm, 1> for ShmState {
+    fn bound_global(&self) -> Result<wl_shm::WlShm, GlobalError> {
+        self.wl_shm().cloned()
```

Comment:
I think you'll need `.map(Clone::clone)` or something like it.

Reply:
Yes, if we plan to leave the minimum rustc version as is.

Reply:
This was stablized in 1.59, I believe we could jump to 1.60 or 1.61.

Winit which depends on the current sctk uses latest rust version.